src/map/atcommand.c

```diff
@@ -8544,6 +8544,9 @@ ACMD(set)
 			data->type = C_STR;
 			data->u.mutstr = aStrdup(str);
 		}
+	} else if (script->str_data[script->search_str(reg)].type == C_PARAM) {
```

Comment:
this whole atcommand needs to be rewritten really; I'll do that when I get some time